backends/gstreamer/lib.rs

```diff
@@ -106,80 +110,80 @@ impl GStreamerBackend {
             return Err(ErrorLoadingPlugins(errors));
         }
 
+        let instances: Arc<
+            Mutex<HashMap<ClientContextId, Vec<(usize, Weak<Mutex<dyn MediaInstance>>)>>>,
+        > = Arc::new(Mutex::new(HashMap::new()));
+
+        let instances_ = instances.clone();
+        let (backend_chan, recvr) = mpsc::channel();
+        thread::Builder::new()
+            .name("GStreamerBackend ShutdownThread".to_owned())
+            .spawn(move || {
+                match recvr.recv().unwrap() {
+                    BackendMsg::Shutdown(context_id, muteable_id) => {
+                        if let Some(vec) = instances_.lock().unwrap().get_mut(&context_id) {
+                            vec.retain(|m| m.0 != muteable_id);
+                            if vec.is_empty() {
+                                instances_.lock().unwrap().remove(&context_id);
+                            }
+                        }
+                    }
+                };
+            })
```

Comment:
how do we stop this thread?

Reply:
The thread dies as soon as it processes the `BackendMsg::Shutdown` channel message.